sdk-api-src/content/winperf/nc-winperf-pm_collect_proc.md

```diff
@@ -90,29 +88,29 @@ If this function fails, set *lpcbTotalBytes* to zero.
 
 Type: **LPDWORD**
 
-Set *lpNumObjectTypes* to the number of object [types](ns-winperf-perf_object_type.md) (not [instances](ns-winperf-perf_instance_definition.md)) written to the *pData* buffer.
+Set *lpNumObjectTypes* to the number of object [types](ns-winperf-perf_object_type.md) (not [instances](ns-winperf-perf_instance_definition.md)) written to the *pData* buffer (where *pData* refers to the pointer pointed to by *lppData*).
 
 If this function fails, set *lpNumObjectTypes* to zero.
 
 > [!NOTE]
-> This parameter is annotated as both *In* and *Out*. Actually the pointee (the value behind this pointer) is never used as an input, and there are no contracts about its content upon function callback invokation: it is best to assume that the pointee is uninitialized and contains garbage data.
+> This parameter is annotated as both *In* and *Out*, however this parameter should not be used as input. 
 
 ## -returns
 
-Return one of the following return values only.
+One of the following values:
 
 | Return code | Description |
 |-------------|-------------|
-| **ERROR_MORE_DATA** | The size of the *pData* buffer as specified by *lpcbTotalBytes* is not large enough to store the data. Leave *pData* unchanged, and set *lpcbTotalBytes* and *lpNumObjectTypes* to zero. No attempt is made to indicate the required buffer size, because this can change before the next call. |
+| **ERROR_MORE_DATA** | The size of the *pData* buffer (where *pData* refers to the pointer pointed to by *lppData*) as specified by *lpcbTotalBytes* is not large enough to store the data. Leave *pData* unchanged, and set *lpcbTotalBytes* and *lpNumObjectTypes* to zero. No attempt is made to indicate the required buffer size, because this can change before the next call. |
 | **ERROR_SUCCESS** | Return this value in all cases other than the **ERROR_MORE_DATA** case, even if no data is returned or an error occurs. To report errors other than insufficient buffer size, use the Application Event Log. |
 
 ## -remarks
 
-If the requested objects specified in the *lpValueName* parameter do not correspond to any of the object indexes that your performance DLL supports, leave the *pData* parameter unchanged, and set the *lpcbTotalBytes* and *lpNumObjectTypes* parameters to zero. This indicates that no data was returned.
+If the requested objects specified in the *lpValueName* parameter do not correspond to any of the object indexes that your performance DLL supports, leave the *pData* parameter (where *pData* refers to the pointer pointed to by *lppData*) unchanged, and set the *lpcbTotalBytes* and *lpNumObjectTypes* parameters to zero. This indicates that no data was returned.
```

Comment:
I feel uneasy about splitting the sub-sentence like that. May I suggest the following?
 
```suggestion
If the requested objects specified in the *lpValueName* parameter do not correspond to any of the object indexes that your performance DLL supports, leave the *pData* parameter unchanged (where *pData* refers to the pointer pointed to by *lppData*), and set the *lpcbTotalBytes* and *lpNumObjectTypes* parameters to zero. This indicates that no data was returned.
```